android/src/main/kotlin/com/example/user_location_plugin/UserLocationPlugin.kt

```diff
@@ -0,0 +1,131 @@
+package com.example.user_location_plugin
+
+import android.Manifest
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import androidx.annotation.NonNull
+import androidx.core.app.ActivityCompat.requestPermissions
+import androidx.core.content.ContextCompat
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import io.flutter.plugin.common.EventChannel
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry
+
+class UserLocationPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
+    PluginRegistry.ActivityResultListener, PluginRegistry.RequestPermissionsResultListener {
+
+    private lateinit var channel: MethodChannel
+    private var context: Context? = null
+    private var activity: Activity? = null
+
+    private var permissionEventChannel: EventChannel? = null
+    private var permissionEventSource: EventChannel.EventSink? = null
+    private var permissionStreamHandler: EventChannel.StreamHandler =
+        object : EventChannel.StreamHandler {
+            override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
+                permissionEventSource = events
+            }
+
+            override fun onCancel(arguments: Any?) {
+                permissionEventSource = null
+            }
+        }
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        context = flutterPluginBinding.applicationContext
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "user_location_plugin")
+        channel.setMethodCallHandler(this)
+        permissionEventChannel =
+            EventChannel(flutterPluginBinding.binaryMessenger, "permission_event_channel")
+        permissionEventChannel?.setStreamHandler(permissionStreamHandler)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            "getPlatformVersion" -> getPlatformVersion(result)
```

Comment:
lets have these constants on a separate file, it would be more organized

Reply:
i created a companion object instead, because constants are not used in others files, it's ok?